install/kubernetes/cilium/templates/cilium-configmap.yaml

```diff
@@ -814,10 +814,10 @@ data:
 {{- if hasKey .Values.k8s "requireIPv6PodCIDR" }}
   k8s-require-ipv6-pod-cidr: {{ .Values.k8s.requireIPv6PodCIDR | quote }}
 {{- end }}
-{{- if and .Values.endpointRoutes .Values.endpointRoutes.enabled }}
+{{- if hasKey .Values.endpointRoutes "enabled" }}
   enable-endpoint-routes: {{ .Values.endpointRoutes.enabled | quote }}
```

Comment:
It seems the EKS CI failure is actually legitimate. `endpointRoutes` defaults to false, but `ipam:eni` (used by ci-eks) manually sets `enable-endpoint-routes:true` separately: https://github.com/cilium/cilium/blob/d9ea2182400c4262d5bed90001d5763924d29e3a/install/kubernetes/cilium/templates/cilium-configmap.yaml#L504

This means this change breaks EKS. Thinking about this some more, we probably only use the the `hasKey` pattern for booleans that default to true.

Reply:
Hmm, maybe we shouldn't merge this then. I'm all for simplifying code where it makes sense, but I'm worried that someone will copy this pattern blindly in the future and end up with the wrong result.

Reply:
@gandro Yes I was thinking the same, i will Update the PR based on changing only k8sNetworkPolicy and drop the rest

Reply:
Awesome, thank  you!